### ISSUE TITLE
ccall + ASYNCIFY promise support

### ISSUE THREAD

**Issue by curiousdannii** [emscripten-core/emscripten#9029]
#6586 added support for returning a promise from `ccall` when running the Emterpreter. Support should also be added for the new ASYNCIFY mode.

**Comment by AjayP13** [comment]
@kripken Happy to contribute a PR for this as I recently opened another issue where this was causing problems (#9279) if you could point me in the right direction. It looks like it was tagged as `good first issue`. I have some sense of what the old `ccall` looked like for Emterpreter's Async. It seems like all I would have to do is figure out some way to get a callback for when Asyncify's execution is completely done executing (parallel to Emterpreter's `asyncFinalizers`) and resolve a Promise with the return value at that time?

**Comment by AjayP13** [comment]
Was able to figure it out. This issue can be closed once #9280 lands.
